File: techila/examples/R/Tutorial/4_multiplefunctions/run_multi_function.r
# Copyright 2010-2013 Techila Technologies Ltd.

# This function contains the Local Control Code, which will create the
# computational Project. The value of the input argument will
# determine which function will be executed in the computational Jobs.
#
# Usage:
# source("run_multi_function.r")
# result <- run_multi_function(funcname)
# Example:
# result <- run_multi_function("function1")

run_multi_function <- function(funcname) {

  # Load the techila library
  library(techila)

  # Create the computational Project with the peach function.
  result <- peach(funcname = funcname, # Executable function determined by the input argument of 'run_multi_function'
                  files = list("multi_function_dist.r"), # The R-script that will be sourced on Workers
                  peachvector = 1:1, # Set the number of Jobs to one (1)
                  sdkroot = "../../../..") # Location of the techila_settings.ini file

  # Convert the results to numeric format and display them.
  print(as.numeric(result))
}

File: techila/examples/python/Tutorial/4_multiplefunctions/run_multi_function.py
# Copyright 2012-2013 Techila Technologies Ltd.

# This function contains the Local Control Code, which will create the
# computational Project. The value of the input argument will
# determine which function will be executed in the computational Jobs.
#
# Usage:
# execfile('run_multi_function.py')
# result = run_multi_function(funcname)
# Example:
# result = run_multi_function('function1')

def run_multi_function(funcname):

    # Load the techila library
    import techila

    # Create the computational Project with the peach function.
    result = techila.peach(funcname = funcname, # Executable function determined by the input argument of 'run_multi_function'
                           files = ['multi_function_dist.py'], # The Python-script that will be evaluated on Workers
                           peachvector = [1], # Set the number of Jobs to one (1)
                           )

    print(result)
    return(result)
